Makefile

```diff
@@ -1263,44 +1276,175 @@ checkstack: tools/checkstack$(EXE)
+ocamldep_MODULES = tools/ocamldep
+
+tools/ocamldep$(EXE): OC_BYTECODE_LDFLAGS += -compat-32
+
+# The profiler
+
+ocamlprof_LIBRARIES =
+ocamlprof_MODULES = \
+  config build_path_prefix_map misc identifiable numbers arg_helper \
+  local_store load_path clflags terminfo warnings location longident \
+  docstrings syntaxerr ast_helper camlinternalMenhirLib parser pprintast \
+  lexer parse ocamlprof
+
+ocamlcp_ocamloptp_MODULES = \
+  config build_path_prefix_map misc profile warnings identifiable numbers \
+  arg_helper local_store load_path clflags terminfo location ccomp compenv \
+  main_args ocamlcp_common
+
+ocamlcp_LIBRARIES =
+ocamlcp_MODULES = $(ocamlcp_ocamloptp_MODULES) ocamlcp
+
+ocamloptp_LIBRARIES =
+ocamloptp_MODULES = $(ocamlcp_ocamloptp_MODULES) ocamloptp
+
+# To help building mixed-mode libraries (OCaml + C)
+ocamlmklib_LIBRARIES =
+ocamlmklib_MODULES = config build_path_prefix_map misc ocamlmklib
+
+# To make custom toplevels
+
+ocamlmktop_LIBRARIES =
+ocamlmktop_MODULES = \
+  config build_path_prefix_map misc identifiable numbers arg_helper \
+  local_store load_path clflags profile ccomp ocamlmktop
+
+# Reading cmt files
+
+ocamlcmt_LIBRARIES = $(addprefix compilerlibs/ocaml,common bytecomp)
+ocamlcmt_MODULES = tools/ocamlcmt
+
+# The bytecode disassembler
+
+dumpobj_LIBRARIES = $(addprefix compilerlibs/ocaml,common bytecomp)
+dumpobj_MODULES = $(addprefix tools/,opnames dumpobj)
+
+make_opcodes = tools/make_opcodes$(EXE)
+
+make_opcodes_LIBRARIES =
+make_opcodes_MODULES = tools/make_opcodes
+
+tools/opnames.ml: runtime/caml/instruct.h $(make_opcodes)
+	$(NEW_OCAMLRUN) $(make_opcodes) -opnames < $< > $@
+
+clean::
+	rm -f $(addprefix tools/,opnames.ml make_opcodes.ml)
+
+beforedepend:: $(addprefix tools/,opnames.ml make_opcodes.ml)
+
+# Display info on compiled files
+
+ocamlobjinfo_LIBRARIES = \
+  $(addprefix compilerlibs/ocaml,common bytecomp middleend)
+ocamlobjinfo_MODULES = tools/objinfo
+
+# Scan object files for required primitives
+
+primreq_LIBRARIES = $(addprefix compilerlibs/ocaml,common bytecomp)
+primreq_MODULES = tools/primreq
+
+# Copy a bytecode executable, stripping debug info
+
+stripdebug_LIBRARIES = \
+  $(addprefix compilerlibs/ocaml,common bytecomp)
+stripdebug_MODULES = tools/stripdebug
+
+# Compare two bytecode executables
+
+cmpbyt_LIBRARIES = $(addprefix compilerlibs/ocaml,common bytecomp)
+cmpbyt_MODULES = tools/cmpbyt
+
+# Scan latex files, and run ocaml code examples
+
+ocamltex_LIBRARIES = \
+  $(addprefix compilerlibs/ocaml,common bytecomp toplevel) \
+  $(addprefix otherlibs/,str/str unix/unix)
+ocamltex_MODULES = tools/ocamltex
+
+# ocamltex uses str.cma and unix.cma and so must be compiled with
+# $(ROOTDIR)/ocamlc rather than with $(ROOTDIR)/boot/ocamlc since the boot
+# compiler does not necessarily have the correct shared library
+# configuration.
+$(ocamltex): CAMLC = $(OCAMLRUN) $(ROOTDIR)/ocamlc$(EXE) $(STDLIBFLAGS)
```

Comment:
I think it would be worth adding a comment to note that this also applies to the compilation of `ocamltex.ml` and `ocamltex.mli` (because they are dependencies) as that's both important and also a subtle part of how GNU make works